python/deltalake/table.py

```diff
@@ -746,6 +746,31 @@ def delete(self, predicate: Optional[str] = None) -> Dict[str, Any]:
         metrics = self._table.delete(predicate)
         return json.loads(metrics)
 
+    def repair(self, dry_run: bool = True) -> Dict[str, Any]:
```

Comment:
I wonder if we should make `dry_run` `False` by default. It's not a dangerous operation right?

Reply:
It's not a dangerous operation, I just set it to True for consistency because with Vacuum we also do dry runs first.

Reply:
Let's make it `True` by default. `Vacuum` is false because it is actually dangerous; you could accidentally delete files you meant to keep. This one is reversible (IIUC) using restore.

Reply:
I think you meant the other way around. :) False by default 😄

Reply:
@wjones127 can we merge this?